### PR TITLE
feat(nginx): refactor + ipv6

### LINKED ISSUE #1703: [Feature]: Support IPv6
### Discussed in https://github.com/immich-app/immich/discussions/1610

<div type='discussions-op-text'>

<sup>Originally posted by **pr0ton11** February  7, 2023</sup>
### Feature detail

Hi

I am using the Immich Helm Chart in my IPv6 only Kubernetes cluster. Unfortunatly, Immich does not seem to listen to IPv6 at all, making it currently unusable for me.

Is there a parameter where I could tell Immich to bind / listen to an IPv6 address instead?

Thanks you

### Platform

Server</div>

### LINKED ISSUE #1723: [BUG] Can't run immich reverse proxy with IPv6 disabled
### The bug

The reverse proxy nginx container will not run if the host has IPv6 disabled. We can solve this by either having an environment variable to disable IPv6 support, or just try to run with IPv6 and if that fails, try IPv4.

### The OS that Immich Server is running on

N/A

### Version of Immich Server

1.46

### Version of Immich Mobile App

N/A

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
1. Disable IPv6 on host machine
2. Try to run immich 1.46
3. Nginx container will fail
```

### Additional information

_No response_

### LINKED ISSUE #1753: [BUG] 502 Bad Gateway after restarting the Browser (Firefox, Gnome Web)
### The bug

Since upgrading to 1.47.2 from 1.46.0 I get a 502 Bad Gateway when I try to open Immich. Deleting Cockies and Website data solves the issue until it appears again.

### The OS that Immich Server is running on

Docker on Debian Host

### Version of Immich Server

1.47.2

### Version of Immich Mobile App

1.47.0

### Platform with the issue

- [ ] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
version: "3.3"

volumes:
  immich_data:
    external: true

services:
  server:
    image: altran1502/immich-server:release
    entrypoint: ["/bin/sh", "./start-server.sh"]
    volumes:
      - immich_data:/usr/src/app/upload
    environment:
      - NODE_ENV=production
      - DB_HOSTNAME=postgres
      - DB_USERNAME=postgres
      - DB_PASSWORD=${DB_PASSWORD}
      - DB_DATABASE_NAME=immichdb
      - REDIS_HOSTNAME=redis
      - JWT_SECRET=${JWT_SECRET}
      - ENABLE_MAPBOX=false
      - MAPBOX_KEY=${MAPBOX_KEY}
      - VITE_LOGIN_PAGE_MESSAGE="Immich Instance des Koyax Project"
      - ENABLE_MAPBOX=true
    env_file:
      - env
    hostname: immich-server
    depends_on:
      - redis
      - postgres
    restart: unless-stopped

  microservices:
    image: altran1502/immich-server:release
    entrypoint: ["/bin/sh", "./start-microservices.sh"]
    volumes:
      - immich_data:/usr/src/app/upload
    environment:
      - NODE_ENV=production
      - DB_HOSTNAME=postgres
      - DB_USERNAME=postgres
      - DB_PASSWORD=${DB_PASSWORD}
      - DB_DATABASE_NAME=immichdb
      - REDIS_HOSTNAME=redis
      - JWT_SECRET=${JWT_SECRET}
      - ENABLE_MAPBOX=false
      - MAPBOX_KEY=${MAPBOX_KEY}
      - VITE_LOGIN_PAGE_MESSAGE="Immich Instance des Koyax Project"
      - ENABLE_MAPBOX=true
    env_file:
      - env
    depends_on:
      - redis
      - postgres
    restart: unless-stopped
    hostname: immich-microservices
    deploy:
      resources:
        limits:
          cpus: '4.0'
          memory: 4G

  machine-learning:
    image: altran1502/immich-machine-learning:release
    entrypoint: ["/bin/sh", "./entrypoint.sh"]
    volumes:
      - immich_data:/usr/src/app/upload
    environment:
      - NODE_ENV=production
      - DB_HOSTNAME=postgres
      - DB_USERNAME=postgres
      - DB_PASSWORD=${DB_PASSWORD}
      - DB_DATABASE_NAME=immichdb
      - REDIS_HOSTNAME=redis
      - JWT_SECRET=${JWT_SECRET}
      - ENABLE_MAPBOX=false
      - MAPBOX_KEY=${MAPBOX_KEY}
      - VITE_LOGIN_PAGE_MESSAGE="Immich Instance des Koyax Project"
      - ENABLE_MAPBOX=true
    env_file:
      - env
    depends_on:
      - postgres
    restart: unless-stopped
    hostname: immich-machine-learning

  web:
    image: altran1502/immich-web:release
    entrypoint: ["/bin/sh", "./entrypoint.sh"]
    environment:
      - DB_HOSTNAME=postgres
      - DB_USERNAME=postgres
      - DB_PASSWORD=${DB_PASSWORD}
      - DB_DATABASE_NAME=immichdb
      - REDIS_HOSTNAME=redis
      - JWT_SECRET=${JWT_SECRET}
      - ENABLE_MAPBOX=false
      - MAPBOX_KEY=${MAPBOX_KEY}
      - VITE_LOGIN_PAGE_MESSAGE="Immich Instance des Koyax Project"
      - ENABLE_MAPBOX=true
    env_file:
      - env
    restart: unless-stopped
    hostname: immich-web

  redis:
    image: redis:6
    restart: unless-stopped
    hostname: immich_redis

  postgres:
    image: postgres:14
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
      PG_DATA: /var/lib/postgresql/data
    env_file:
      - env
    volumes:
      - /opt/immich-postgres:/var/lib/postgresql/data
    restart: unless-stopped
    hostname: immich_postgres

  proxy:
    image: altran1502/immich-proxy:release
    ports:
      - 2283:8080
    depends_on:
      - server
    restart: unless-stopped
```

### Your .env content

```Shell
OAUTH_ENABLED=true
OAUTH_ISSUER_URL=https://account.koyax.org/application/o/immich/
OAUTH_CLIENT_ID=${OPEN_ID_ID}
OAUTH_CLIENT_SECRET=${OPEN_ID_SECRET}
OAUTH_BUTTON_TEXT=Login mit Koyax ID
```

### Reproduction steps

```bash
1. Login to Immich
2. Watch you precious assets
3. Close the browser
4. Retry to open the immich page
5. 502
...
```

### Additional information

* I use Authentik for Login with OIDC
* Sensitive Information is set in Portainer

### PR DESCRIPTION
**Pull request by michelheusschen**
Refactored the nginx Docker file to include the default entrypoint files for env substitution and tuning `worker_processes`. The entrypoint for dynamically enabling IPv6 has been copied from the default nginx Docker image, with the file checksum disabled and listening port modified. This should only enable IPv6 in environments that support it with IPv4 always enabled. Fixes #1703 and fixes #1723.

Minor adjustments to `gzip_types` by removing text/html which is included by default and caused a warn log message.

### CODE REVIEW

**Comment by zackpollard** [comment]
Hey, for #1753 we had a discussion internally and would rather disable svelte sending the javascript in the headers entirely. The optimisation from this is likely very little and it will require changes beyond just our reverse proxy for people using their own custom setups. 
image
This is the advice from the creator of svelte for this, if you are willing to include that in your PR and revert the buffer change, that'd be great.

**Comment by michelheusschen** [comment]
Changes to proxy buffering have been reverted. I will open a new PR for removing the Link header.

File: nginx/10-listen-on-ipv6-by-default.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+grep -q "listen  \[::]\:8080;" /$DEFAULT_CONF_FILE && { entrypoint_log "$ME: info: IPv6 listen already enabled"; exit 0; }
+
+if [ -f "/etc/os-release" ]; then
+    . /etc/os-release
+else
+    entrypoint_log "$ME: info: can not guess the operating system"
+    exit 0
+fi
+
+# entrypoint_log "$ME: info: Getting the checksum of /$DEFAULT_CONF_FILE"
```
Comment (zackpollard) [inline nginx/10-listen-on-ipv6-by-default.sh:39]: Should probably just remove this section entirely instead of leaving it commented out?

**Review by zackpollard — APPROVED** [review]
LGTM, one small comment but nice changes here. I haven't tested them myself but the changes look good.

File: nginx/10-listen-on-ipv6-by-default.sh (hunk 1 above)
Comment (michelheusschen) [reply nginx/10-listen-on-ipv6-by-default.sh]: Yeah that doesn't need to be there anymore, removed it!

**Pull request merged**